source/views/streaming/streams/list.js

```diff
@@ -128,8 +138,8 @@ export class StreamListView extends React.PureComponent {
         ItemSeparatorComponent={ListSeparator}
         keyExtractor={this.keyExtractor}
         style={styles.listContainer}
-        refreshing={this.props.loading}
-        onRefresh={this.props.onRefresh}
+        refreshing={this.state.refreshing}
+        onRefresh={this.getData}
```

Comment:
I like to define a second function that calls my `getData` function, but does the timing things separately. That way, my initial component mounting isn't delayed.

I usually call that function `refresh`, and it does the `setState{refreshing:true}`, `getData()`, `setState{refreshing:false}`. For example, the one in Calendar: 

https://github.com/StoDevX/AAO-React-Native/blob/e04aaea8810ce1ffaa1c7693a4e401add7bc5f9c/source/views/calendar/calendar-google.js#L97-L110

Reply:
Great idea on making it easier to cleaner. Thanks.